dialogflow/context_management.rb

```diff
@@ -0,0 +1,116 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require 'securerandom'
+
+def list_contexts project_id:, session_id:
+  # [START dialogflow_list_contexts]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "existing_session_id"
+  
+  require "google/cloud/dialogflow"
+
+  contexts_client = Google::Cloud::Dialogflow::Contexts.new
+  session_path = contexts_client.class.session_path project_id, session_id
+
+  contexts = contexts_client.list_contexts(session_path)
+
+  puts "Contexts for session #{session_path}:\n"
+  contexts.each do |context|
+    puts "Context name:   #{context.name}"
+    puts "Lifespan count: #{context.lifespan_count}"
+    if !context.parameters.nil?
+      puts "Fields:"
+      context.parameters.fields.each do |field, value|
+        if value.string_value
```

Comment:
... what if it's not a `string_value`?

Could `value` have like... a `value.int32_value` or `value.bool_value` or anything?

Is there a way to just print out the value (and it'll get `toString()`'d, `#to_s` in Ruby), regardless of value type?

Reply:
If the proto object doesn't allow this and doesn't implement a useful toString, then, you COULD do this... (I'm guessing the types)... which would be more educational to users reading it IMO *(because I found .string_value to be really confusing)*

```ruby
context.parameters.fields.each do |field, value|
  # The value of the field is accessed via different fields, depending on its value type
  field_value = [
    value.string_value, value.bool_value, value.int_value, int.float_value
  ].first &:present?

  puts "#{field} = #{field_value}"
end

^^^ Nevermind, this is gross, blah, why is it like this?
```

I don't really like it though... is `value` an `Any` and that's why it has a `string_value` instead of a proper, named field?

Reply:
in this case `string_value` is the name of a field of the `value` message: 

https://github.com/dialogflow/dialogflow-nodejs-client-v2/blob/master/protos/google/cloud/dialogflow/v2beta1/context.proto#L88

https://github.com/google/protobuf/blob/master/src/google/protobuf/struct.proto#L52

https://github.com/google/protobuf/blob/master/src/google/protobuf/struct.proto#L63

It is very unfortunate that the we decided to use things like `Struct` and `Value` messages here.  Indeed `Value` might have other values than `string_value`, but I only remember seeing `string_value` populated by dialogflow agents; it is possible that the dialogflow API defers to the user to type cast, even if some of the values seem useful, e.g. `number_value`.